tracer/src/Datadog.Trace/Activity/Handlers/DefaultActivityHandler.cs

```diff
@@ -130,15 +144,16 @@ public void ActivityStopped<T>(string sourceName, T activity)
                         return;
                     }
 
-                    if (ActivityMappingById.TryRemove(activity.Id, out ActivityMapping value) && value.Scope?.Span is not null)
+                    // first we look for the normal ID, then the parent
```

Comment:
I don't understand the comment TBH (it's Friday sorry)

Reply:
I think this is related to the "differential" registration, but I'm not sure why we didn't register the ID of this activity

Reply:
Ah yeah good catch
I deleted a branch here but need it back now, basically since I'm storing the `ParentId` for the `ActivityContext` the `activity.Id` check will fail, so if it does fail then the idea was to check for the `activity.ParentId` which was stored earlier for that `ActivityContext`